docs/src/main/asciidoc/_elytron/Credential_Store.adoc

```diff
@@ -0,0 +1,566 @@
+----
+
+The following code illustrates how a couple of different credential types can be added to a credential store: -
+
+[source,java,options="nowrap"]
+----
+Password clearPassword = ClearPassword.createRaw(ClearPassword.ALGORITHM_CLEAR, "ExamplePassword".toCharArray());
+credentialStore.store("clearPassword", new PasswordCredential(clearPassword));
+
+KeyGenerator keyGenerator = KeyGenerator.getInstance("AES");
+keyGenerator.init(256);
+SecretKey secretKey = keyGenerator.generateKey();
+credentialStore.store("secretKey", new SecretKeyCredential(secretKey));
+----
+
+These credentials can then be obtained again from the store: -
+
+[source,java,options="nowrap"]
+----
+Password password = credentialStore.retrieve("clearPassword", PasswordCredential.class).getPassword();
+SecretKey secretKey = credentialStore.retrieve("secretKey", SecretKeyCredential.class).getSecretKey();
+----
+
+NOTE: As the type is specified when retrieving a credential it is possible to store multiple credentials under the same alias.
+
+Please use the published javadoc for more information in relation to the APIs and the credential types supported within WildFly Elytron.
+
+== Migrating Existing Vaults
+
+If migrating from a prior version of the application server it is possible that you already are making use of a PicketBox vault for the storage of clear text passwords, the tooling provided can be used to convert the vault to the format used by the `KeyStoreCredentialStore`.
+
+Within the WildFly Elytron command line tool an additional command `vault` is available specifically for the conversion of legacy vaults to a credential store, a complete vault can be converted to a credential store with the following command: - 
+
+The following command can be used to convert a single entry from a vault to a credential store: -
+
+[source,options="nowrap"]
+----
+]$ bin/elytron-tool.sh vault --enc-dir standalone/configuration/vault --keystore standalone/configuration/vault.keystore --iteration 44 --salt 00000000 --alias vault \ 
+    --location standalone/configuration/newcredstore.cs 
+Vault password: 
+Confirm vault password: 
+Vault (enc-dir="standalone/configuration/vault";keystore="standalone/configuration/vault.keystore") converted to credential store "standalone/configuration/newcredstore.cs"
+----
+
+When executing this command the destination credential store must not already exist.  The password used for the credential store will be the password originally used for the vault.
+
+Entries stored within the vault would have been stored specifying a "block" and "alias" value, within the credential store the new alias will be `block::alias`.
+
+[[Custom_CredentialStore]]
+== Custom Credential Store
+
+It is also possible to provide custom credential store implementations, overall the pattern to implementing a custom credential store is very similar to the pattern that would be followed to implement a custom credential store.
```

Comment:
s/implementations, overall/implementations. Overall/g

Reply:
The second part of this doesn't make sense since it says  "the way to do foo is just like the way to do foo." One of the 'foo's should be a 'bar'?